logbook-json/src/main/java/org/zalando/logbook/json/PrimitiveJsonPropertyBodyFilter.java

```diff
@@ -63,25 +65,31 @@
     @With(PRIVATE)
     private final Predicate<String> predicate;
 
-    private final String replacement;
+    private final BinaryOperator<String> replacement;
 
     private static Pattern pattern(final String value) {
-        return compile("(?<key>\"(?<property>" + STRING_VALUE_PATTERN + ")\"\\s*:\\s*)(" + value + "|null)");
+        return compile("(?<key>\"(?<property>" + STRING_VALUE_PATTERN + ")\"\\s*:\\s*)(?<propertyValue>" + value + "|null)");
     }
 
     static BodyFilter replaceString(
             final Predicate<String> predicate, final String replacement) {
-        return create(STRING, predicate, quote(replacement));
+        return create(STRING, predicate, new StaticParameterReplacementOperator<>(replacement, true));
     }
 
     static BodyFilter replaceNumber(
             final Predicate<String> predicate, final Number replacement) {
-        return create(NUMBER, predicate, String.valueOf(replacement));
+        return create(NUMBER, predicate, new StaticParameterReplacementOperator<>(replacement, false));
     }
 
     static BodyFilter replacePrimitive(
             final Predicate<String> predicate, final String replacement) {
-        return create(PRIMITIVE, predicate, quote(replacement));
```

Comment:
Interesting. I didn't realize that we always quoted here. Means there is no way to replace booleans and numbers because they will become strings.